core/trino-main/src/main/java/io/trino/operator/TableDeleteOperator.java

```diff
@@ -13,28 +13,19 @@
  */
 package io.trino.operator;
 
-import com.google.common.collect.ImmutableList;
 import io.trino.Session;
 import io.trino.metadata.Metadata;
 import io.trino.metadata.TableHandle;
-import io.trino.spi.Page;
-import io.trino.spi.PageBuilder;
-import io.trino.spi.block.BlockBuilder;
-import io.trino.spi.type.Type;
 import io.trino.sql.planner.plan.PlanNodeId;
 
-import java.util.List;
 import java.util.OptionalLong;
 
 import static com.google.common.base.Preconditions.checkState;
-import static io.trino.spi.type.BigintType.BIGINT;
 import static java.util.Objects.requireNonNull;
 
 public class TableDeleteOperator
-        implements Operator
+        extends TableMetadataOperator
```

Comment:
Let's put introduction of `TableMetadataOperator` and `TableMetadataOperationNode` in separate commit(s)